### PR TITLE
Clean up stonith_admin

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

**Comment by kgaillot** [comment]
For easier review, here is the new help output:
----
% stonith_admin --help
stonith_admin - access the Pacemaker fencing API
Usage: stonith_admin <command> [<options>]
Options:
 -?, --help		Display this text and exit.
 -$, --version		Display version information and exit.
 -V, --verbose		Increase debug output (may be specified multiple times).
 -q, --quiet		Be less descriptive in output.

Device definition commands:
 -R, --register=value	Register the named stonith device. Requires: --agent.
			Optional: any number of --option and/or --env entries.
 -D, --deregister=value	De-register the named stonith device.
 -r, --register-level=value	Register a stonith level for the named target,
			specified as one of NAME, @PATTERN, or ATTR=VALUE.
			Requires: --index and one or more --device entries.
 -d, --deregister-level=value	Unregister a stonith level for the named target,
			specified as for --register-level. Requires: --index.

Queries:
 -l, --list=value	List devices that can terminate the specified host.
			Optional: --timeout.
 -L, --list-registered	List all registered devices. Optional: --timeout.
 -I, --list-installed	List all installed devices. Optional: --timeout.
 -s, --list-targets=value	List the targets that can be fenced by the
			named device. Optional: --timeout.
 -M, --metadata		Show agent metadata. Requires: --agent.
			Optional: --timeout.
 -Q, --query=value	Check the named device's status. Optional: --timeout.

Fencing Commands:
 -F, --fence=value	Fence named host. Optional: --timeout, --tolerance.
 -U, --unfence=value	Unfence named host. Optional: --timeout, --tolerance.
 -B, --reboot=value	Reboot named host. Optional: --timeout, --tolerance.
 -C, --confirm=value	Tell cluster that named host is now safely down.
 -H, --history=value	Show last successful fencing operation for named node
			(or '*' for all nodes). Optional: --timeout, --quiet
			(show only the operation's epoch timestamp),
			--verbose (show all recorded and pending operations).
 -h, --last=value	Indicate when the named node was last fenced.
			Optional: --as-node-id.

Additional Options:
 -a, --agent=value	The agent to use (for example, fence_xvm;
			with --register, --metadata).
 -o, --option=value	Specify a device configuration parameter as NAME=VALUE
			(with --register).
 -e, --env-option=value	Specify a device configuration parameter with the
			specified name, using the value of the
			environment variable of the same name prefixed with
			OCF_RESKEY_ (with --register).
 -T, --tag=value	Identify fencing operations in logs with the specified
			tag; useful when multiple entities might invoke
			stonith_admin (used with most commands).
 -v, --device=value	A device to associate with a given host and
			stonith level (with --register-level).
 -i, --index=value	The stonith level (1-9) (with --register-level,
			--deregister-level).
 -t, --timeout=value	Operation timeout in seconds (default 120;
			used with most commands).
 -n, --as-node-id	(Advanced) The supplied node is the corosync node ID
			(with --last).
     --tolerance=value	(Advanced) Do nothing if an equivalent --fence request
			succeeded less than this many seconds earlier
			(with --fence, --unfence, --reboot).

Report bugs to users@clusterlabs.org
----

File: fencing/admin.c [hunk 1]
```diff
@@ -337,32 +364,23 @@ show_history(stonith_t *st, const char *target, int timeout, int quiet,
...
-        } else if (hp->action == NULL) {
-            action_s = strdup("unknown");
-        } else if (hp->action[0] != 'r') {
-            action_s = crm_concat("turn", hp->action, ' ');
-        } else {
-            action_s = strdup(hp->action);
         }
 
         if (hp->state == st_failed) {
+            action_s = fence_action_str(hp->action);
```
Comment (wenningerk) [inline fencing/admin.c]: didn't see the free for that

File: fencing/admin.c [hunk 2]
```diff
@@ -337,32 +364,23 @@ show_history(stonith_t *st, const char *target, int timeout, int quiet,
...
-            printf("We were able to %s node %s on behalf of %s from %s at %s\n",
-                   action_s, hp->target, hp->client, hp->origin, ctime(&complete));
+            print_fence_event(latest);
+
         } else {
             /* ocf:pacemaker:controld depends on "wishes to" being
              * in this output, when used with older versions of DLM
              * that don't report stateful_merge_wait
              */
+            action_s = fence_action_str(hp->action);
```
Comment (wenningerk) [inline fencing/admin.c]: didn't see the free for that

File: fencing/admin.c [hunk 3]
```diff
@@ -317,6 +317,33 @@ handle_level(stonith_t *st, char *target, int fence_level,
...
+        str = strdup(action);
+    }
+    return str;
+}
+
+static void
+print_fence_event(stonith_history_t *event)
+{
+    char *action_s = fence_action_str(event->action);
+    time_t complete = event->completed;
```
[the remark this answers is not in the document]
Comment (kgaillot) [reply fencing/admin.c]: good question, I just copied-and-pasted, but it's unnecessary

File: fencing/admin.c (hunk 2 above)
Comment (kgaillot) [reply fencing/admin.c]: the free() is after the if-else block

File: fencing/admin.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (kgaillot) [reply fencing/admin.c]: ah, it's necessary because event->completed is an int, and we need a time_t*

[the remark this answers is not in the document]
Comment (kgaillot) [reply fencing/admin.c]: stonith_history_t's completed really should be a time_t, i'll change that for 2.0 ...

File: fencing/admin.c (hunk 2 above)
Comment (wenningerk) [reply fencing/admin.c]: sry - misled by the block without action_s - but as free(NULL) is defined as NOP ...

File: fencing/admin.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (wenningerk) [reply fencing/admin.c]: aah interesting would have though that an implicit cast there  would have created a warning

**Pull request merged**